docs/introduction/why-rtk-is-redux-today.md

```diff
@@ -0,0 +1,220 @@
+
+For example, a reducer function is _just_ a function. Prior to Redux Toolkit, you'd typically write it that reducer with a `switch` statement and manual updates. You'd also probably have hand-written action creators and action type constants along with it:
+
+```js title="Legacy hand-written Redux usage"
+const ADD_TODO = 'ADD_TODO'
+const TODO_TOGGLED = 'TODO_TOGGLED'
+
+export const addTodo = text => ({
+  type: ADD_TODO,
+  payload: { text, id: nanoid() }
+})
+
+export const todoToggled = id => ({
+  type: ADD_TODO,
+  payload: id
+})
+
+export const todosReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADD_TODO:
+      return state.concat({
+        id: action.payload.id,
+        text: action.payload.text,
+        completed: false
+      })
+    case TODO_TOGGLED:
+      return state.map(todo => {
+        if (todo.id !== action.payload) return
+
+        return {
+          ...todo,
+          completed: !todo.completed
+        }
+      })
+    default:
+      return state
+  }
+}
+```
+
+None of this code specifically depends on any API from the `redux` core library. But, this is a lot of code to write. Immutable updates required a lot of hand-written object spreads and array operations, and it was very easy to make mistakes and accidentally mutate state in the process. It was also common, though not strictly required, to spread that same code across multiple files: `actions/todos.js`, `constants/todos.js`, and `reducers/todos.js`.
+
+Additionally, store setup usually required a few dozen lines of setting up commonly used middleware like thunks, hooking up the Redux DevTools Extension support
+
+### What Does Redux Toolkit Do?
+
+While these _were_ the patterns originally shown in the Redux docs, they unfortunately require a lot of very verbose and repetitive code. Most of this boilerplate isn't _necessary_ to use Redux. On top of that, the boilerplate-y code lead to more opportunities to make mistakes.
+
+**We specifically created Redux Toolkit to eliminate the "boilerplate" from hand-written Redux logic, prevent common mistakes, and provide APIs that simplify standard Redux tasks**.
+
+Redux Toolkit starts with two key APIs that simplify the most common things you do in every Redux app:
+
+- `configureStore` sets up a well-configured Redux store with a single function call, including combining reducers, adding the thunk middleware, and setting up the Redux DevTools integration. It also is easier to configure than `createStore`, because it takes named options parameters.
+- `createSlice` lets you write reducers that use [the Immer library](https://immerjs.github.io/immer/) to enable writing "mutating" update syntax that becomes safe and correct immutable updates, with no spreads needed. It also automatically generates action creator functions for each reducer, and generates action type strings internally based on your reducer's names. Finally, it works great with TypeScript.
```

Comment:
I'm stumbling about the "that **becomes** safe" here. I'm sure it is correct, but my brain has problems processing it. Maybe reword?